### PR TITLE
Fix a potential bug in step-15.

### PR DESCRIPTION
**Pull request by bangerth**
This is the underlying issue for what is going wrong in step-77 (#11953): When we set nodes at the boundary to their correct boundary values, we need to make sure that we also update the hanging nodes that are adjacent to these nodes. 

This didn't show up in step-15 because in every step we update the solution, and then call `hanging_node_constraints.distribute()` on the solution somewhere. But it means that we start with a sub-optimal starting guess on every new mesh.

/rebuild

### CODE REVIEW

File: examples/step-15/step-15.cc [hunk 1]
```diff
@@ -430,6 +423,14 @@ namespace Step15
 
     hanging_node_constraints.distribute(current_solution);
```
Comment (marcfehling) [inline examples/step-15/step-15.cc:424]: I think this is obsolete then?

**Review by marcfehling — APPROVED** [review]

File: examples/step-15/step-15.cc (hunk 1 above)
Comment (bangerth) [reply examples/step-15/step-15.cc]: Good point. Fixed now!

**Comment by jppelteret** [comment]
That's frustrating. Maybe in the future we should just combine the Dirichlet and hanging node constraints within a single `AffineConstraints` object? I'd considered doing that for step-72, but then decided that the divergence from step-15 was counter-productive.

**Comment by bangerth** [comment]
Yes, one could have done that as well. I think it doesn't matter much -- it's easy enough to explain what is happening, one just has to have the wherewithal to actually think of the issue.

**Pull request merged**